### ISSUE TITLE
Dialyzer can't parse opaque data types with emojis

### ISSUE THREAD

**Issue by elbrujohalcon** [erlang/otp#5210]
**Describe the bug**
If you have a module with an opaque data type that includes an emoji, dialyzer crashes.

**To Reproduce**
Create these 2 modules in an application…

```erlang
-module(problematic).

-opaque '🐛'() :: bug.
-export_type(['🐛'/0]).

-export(['🔍'/1]).

-spec '🔍'('🐛'()) -> true.
'🔍'(bug) -> true.
```

```erlang
-module(exposer).

-export(['💣'/0]).

'💣'() -> problematic:'🔍'(bug).
```

Then run `DIAGNOSTIC=1 rebar3 dialyzer`

You should see something like…
```erlang
…
src/exposer.erl
Line 5 Column 26: The call problematic:'\x{1F50D}'('bug') does not have an opaque term of type problematic:'🐛'() as 1st argument
===> Setting paths to [plugins,deps]
===> Uncaught error in rebar_core. Run with DIAGNOSTIC=1 to see stacktrace or consult rebar3.crashdump
===> Uncaught error: badarg
===> Stack trace to the error location:
[{re,run,
     [[[["-type ",
         [[[["t",["()"]]," ::"],
           [32,[[["problematic",58,[39,128027,39]],["()"]]]]]]],
        ".\n"]],
      <<"::\\s*(.*)\\.\\n*">>,
      [{capture,all_but_first,list},dotall]],
     [{error_info,#{module => erl_stdlib_errors}}]},
 {dialyzer,pp_type,1,[{file,"dialyzer.erl"},{line,727}]},
 {dialyzer,t,2,[{file,"dialyzer.erl"},{line,700}]},
 {dialyzer,form_expected_without_opaque,2,[{file,"dialyzer.erl"},{line,594}]},
 {dialyzer,message_to_string,3,[{file,"dialyzer.erl"},{line,475}]},
 {dialyzer,format_warning,2,[{file,"dialyzer.erl"},{line,312}]},
 {rebar_prv_dialyzer,'-file_warnings/3-lc$^1/1-0-',1,
                     [{file,"/home/runner/work/rebar3/rebar3/src/rebar_prv_dialyzer.erl"},
                      {line,580}]},
 {rebar_prv_dialyzer,file_warnings,3,
                     [{file,"/home/runner/work/rebar3/rebar3/src/rebar_prv_dialyzer.erl"},
                      {line,581}]}]
```

**Expected behavior**
Just the warning, but no errors.
```
src/exposer.erl
Line 5 Column 26: The call problematic:'\x{1F50D}'('bug') does not have an opaque term of type problematic:'🐛'() as 1st argument
```

**Affected versions**
I tested only on OTP24

**Additional context**
```
Rebar3 report
 version 3.17.0
 generated at 2021-09-18T10:48:57+00:00
=================
Please submit this along with your issue at https://github.com/erlang/rebar3/issues (and feel free to edit out private information, if any)
-----------------
Task: rebar3
Entered as:
  rebar3 dialyzer
-----------------
Operating System: x86_64-apple-darwin20.5.0
ERTS: Erlang/OTP 24 [erts-12.0.2] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]
Root Directory: /Users/fernandobenavides/.kerl/installations/24.0.2
Library directory: /Users/fernandobenavides/.kerl/installations/24.0.2/lib
-----------------
Loaded Applications:
bbmustache: 1.10.0
certifi: 2.6.1
cf: 0.3.1
common_test: 1.20.4
compiler: 8.0.1
crypto: 5.0.2
cth_readable: 1.5.1
dialyzer: 4.4
edoc: 1.0
erlware_commons: 1.5.0
eunit: 2.6.1
eunit_formatters: 0.5.0
getopt: 1.0.1
inets: 7.4
kernel: 8.0.1
providers: 1.8.1
public_key: 1.11
relx: 4.5.0
sasl: 4.1
snmp: 5.9.1
ssl_verify_fun: 1.1.6
stdlib: 3.15.1
syntax_tools: 2.6
tools: 3.5

-----------------
Escript path: /usr/local/bin/rebar3
Providers:
  app_discovery as clean compile compile cover ct cut deps dialyzer do docs edoc escriptize eunit format get-deps hank hank help install install_deps key kiwf kiwf lint list lock new owner path pkgs publish release relup repo report repos retire revert search shell state tar test tree unlock update upgrade upgrade upgrade user version xref
```

**Comment by jhogberg** [comment]
Thanks for your report, I've fixed it in bad25d08e2aa65749a4574463a97fe1658f2daa3 which will be included in OTP-24.2.

**Comment by elbrujohalcon** [comment]
> Thanks for your report, I've fixed it in [bad25d0](https://github.com/erlang/otp/commit/bad25d08e2aa65749a4574463a97fe1658f2daa3) which will be included in OTP-24.2.

# 🎉
